apps/datahub/src/app/record/record-user-feedbacks/record-user-feedbacks.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core'
+import { UserFeedback, UserFeedbackViewModel } from '@geonetwork-ui/common/domain/model/record'
+import { GravatarService } from '@geonetwork-ui/api/repository'
+import { MdViewFacade } from '@geonetwork-ui/feature/record'
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RecordUserFeedbacksService {
+
+  constructor(
+    private mdViewFacade: MdViewFacade,
+    private gravatarService: GravatarService
+  ) {
+  }
+
+  userFeedbacks$ = this.mdViewFacade.userFeedbacks$
+
+  addUserFeedback(userFeedback: UserFeedback) {
+    return this.mdViewFacade.addUserFeedback(userFeedback)
+  }
+
+  loadUserFeedbacks(metadataUuid: string) {
+    return this.mdViewFacade.loadUserFeedbacks(metadataUuid)
+  }
+
+  sortByDateFromNewestToOldest(userFeedbackA: UserFeedback, userFeedbackB: UserFeedback): number {
+    return new Date(userFeedbackB.date).getTime() - new Date(userFeedbackA.date).getTime()
+  }
+
+  sortByDateFromOldestToNewest(userFeedbackA: UserFeedback, userFeedbackB: UserFeedback): number {
+    return new Date(userFeedbackA.date).getTime() - new Date(userFeedbackB.date).getTime()
+  }
+
+  createViewModel(baseUserFeedback: UserFeedback): UserFeedbackViewModel {
+    const userAvatarUrl = this.gravatarService.getProfileIcon(baseUserFeedback.authorUserId?.toString())
+
+    return {
+      ...baseUserFeedback,
+      avatarUrl: userAvatarUrl
```

Comment:
In principle this should be provided by the platform alongside the rest of the feedback. This might require quite a bit of refactoring